portfolio/src/main/webapp/script.js

```diff
@@ -250,3 +251,41 @@ function deleteComment(comment) {
   params.append('id', comment.id);
   fetch('/delete-comments', {method: 'POST', body: params});
 }
+
+/**
+ * Checks if the user is logged in and displays corresponding welcome messages and login/logout buttons.
+ * Also, if user is logged in, enables posting comments and displays set-nickname button.
+ */
+function setupPageByLoginStatus() {
+  fetch('/login').then(response => response.json()).then(loginStatus => {
+    const isLoggedIn = loginStatus.isLoggedIn;
+    const loginOutLink = document.getElementById('login-out');
+    const welcomeElement = document.getElementById('welcome-message');
+
+    if (isLoggedIn) {
+      fetch('/nickname?userId=' + loginStatus.userId).then(response => response.text()).then(displayName => {
+        welcomeElement.innerText = 'Welcome, ' + displayName;
+      });
+
+      document.getElementById('input-form-fieldset').removeAttribute('disabled');
+      document.getElementById('set-nickname-button').style.display = 'block';
+      loginOutLink.href = loginStatus.logoutUrl;
+      loginOutLink.innerText = 'Logout';
+    } else {
+      welcomeElement.innerText = 'Welcome, please log in to post a comment.';
+      
+      document.getElementById('input-form-fieldset').setAttribute('disabled', 'true');
+      document.getElementById('set-nickname-button').style.display = 'none';
+      loginOutLink.href = loginStatus.loginUrl;
+      loginOutLink.innerText = 'Login';
+    }
+  });
+}
+
+/** 
+ * Checks log in status and fetches all the comments to set up comments.html when the page is loading.
+ */
+function loadCommentsPage() {
+  setupPageByLoginStatus()
+  fetchComments(true);
+}
```

Comment:
Just to confirm, does this also actually display the comments? 

If so, you want to update the JS doc to say that part.

Reply:
Ok